cachito/utils.py

```diff
@@ -0,0 +1,174 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import itertools
+import json
+import logging
+import os
+from pathlib import Path
+from typing import Any, Dict, Iterator, List, Tuple, Union
+
+from cachito.errors import CachitoError
+
+log = logging.getLogger(__name__)
+
+
+def package_sort_key(package: Dict[str, Any]) -> Tuple[str, bool, str, str]:
+    """Return the sort key for sorting packages.
+
+    :param package: a mapping representing a package information. It must
+        contain four keys, type, dev, name and version.
+    :type package: dict[str, any]
+    :return: a four-elements tuple as the sort key in the order of type, dev,
+        name and version.
+    :rtype: tuple[str, bool, str, str]
+    """
+    return package["type"], package.get("dev", False), package["name"], package["version"]
+
+
+def sort_packages_and_deps_in_place(packages: List[Dict[str, Any]]) -> None:
```

Comment:
Besides some unit tests, it seems this function is now only being used inside the PackageData class.

If we could work around the tests, it would be possible to make it a private function inside the PackageData class, and this way we'd have less `utils` going around.

We don't need to do this right now, though, it's just a thought.